index.d.ts

```diff
@@ -53,7 +53,7 @@ declare namespace customPlugin {
     getServiceProxy: (options?: InitServiceOptions) => Service,
   }
 
-  interface DecoratedRequest extends fastify.FastifyRequest<http.IncomingMessage> {
```

Comment:
Why `<http.IncomingMessage>` is removed? Also for the others. It is infer by typescript 4?

Reply:
From what I understood reading the different materials regarding the changes between v2 and v3, with Fastify v3 they changed the definition of FastifyRequest and FastifyReply objects. In fact, they are no more based on the types of `http` package, but they rather use their own ones.

At the moment my knowledge in Typescript is still rough, but observing Fastify code I can confirm the changes:

**Request**
v2: https://github.com/fastify/fastify/blob/2.x/fastify.d.ts#L142
v3: https://github.com/fastify/fastify/blob/bfd8cee4e3eed6553312c35bb986bae2229866cc/types/request.d.ts#L16

**Reply**
v2: https://github.com/fastify/fastify/blob/2.x/fastify.d.ts#L171
v3: https://github.com/fastify/fastify/blob/bfd8cee4e3eed6553312c35bb986bae2229866cc/types/reply.d.ts#L15

Maybe `<http.IncomingMessage>` and `<http.ServerResponse>` can be exchanged with the proper corresponding types according to Fastify v3 definitions.

Reply:
Hi @davidebianchi
I checked whether the `<http.IncomingMessage>` can be replaced with the correct new type. In theory it should be possible to exchange that type with `<RouteGeneric, RawServer, RawRequest>` as I discussed in the previous comment.

However, the issue is that those three types are not exported outside the fastify library, which prevents us to actually set them in our types definition.
As a consequence, is it acceptable to extends only the `fastify.FastifyRequest` without specifying the generics types or should we find another alternative?